### PR TITLE
clp: migrate to proto

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

Migrate clp module to proto

- [x] add queries, both grpc and grpc-gateway
- [x] add msg_server
- [x] migrate storage to proto
- [x] migrate cli to new api
- [x] register grpc txs

### CODE REVIEW

File: proto/sifnode/clp/v1/gensis.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
```
Comment (timlind) [inline proto/sifnode/clp/v1/gensis.proto:1]: file name typo

File: proto/sifnode/clp/v1/querier.proto [hunk 2]
```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+package sifnode.clp.v1;
+
+import "gogoproto/gogo.proto";
+import "sifnode/clp/v1/types.proto";
+
+option go_package = "github.com/sifchain/sifnode/x/clp/types";
+
+service Query {
+  rpc QueryGetPool(QueryGetPoolRequest) returns (QueryGetPoolResponse);
```
Comment (timlind) [inline proto/sifnode/clp/v1/querier.proto:10]: I think the guideline is to leave off "query" in the roc name, unless this is for compatibility?

File: x/oracle/types/types.pb.go [hunk 3]
```diff
@@ -27,24 +27,28 @@ const _ = proto.GoGoProtoPackageIsVersion3 // please upgrade the proto package
...
 
 const (
-	StatusText_UNKNOWN_STATUS_TEXT StatusText = 0
-	StatusText_PEDNING_STATUS_TEXT StatusText = 1
-	StatusText_SUCCESS_STATUS_TEXT StatusText = 2
-	StatusText_FAILED_STATUS_TEXT  StatusText = 3
+	// Default value
+	StatusText_STATUS_TEXT_UNSPECIFIED StatusText = 0
+	// Pending status
+	StatusText_STATUS_TEXT_PEDNING StatusText = 1
```
Comment (timlind) [inline x/oracle/types/types.pb.go:33]: Still got a type in the oracle proto PEDNING

Comment (marbar3778) [reply x/oracle/types/types.pb.go:33]: oh whoops thanks.

File: proto/sifnode/clp/v1/querier.proto [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+import "gogoproto/gogo.proto";
+import "sifnode/clp/v1/types.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "google/api/annotations.proto";
+
+option go_package = "github.com/sifchain/sifnode/x/clp/types";
+
+service Query {
+  rpc QueryGetPool(PoolReq) returns (PoolRes) {
```
Comment (timlind) [inline proto/sifnode/clp/v1/querier.proto:12]: "QueryGet" redundancy in name.

**Review by timlind — APPROVED** [review]

**Review by iverc — APPROVED** [review]
Looks good! 🚀 Some cleanup/refactoring is needed, but could be done in the final merge!

File: proto/sifnode/clp/v1/querier.proto (hunk 4 above)
Comment (iverc) [reply proto/sifnode/clp/v1/querier.proto:12]: yeah, there are still changes to be made, but let's put everything together and do a final round of cleanup/refactoring

Comment (marbar3778) [reply proto/sifnode/clp/v1/querier.proto]: fixed,

**Comment by marbar3778** [comment]
> Looks good! 🚀 Some cleanup/refactoring is needed, but could be done in the final merge!

want to comment on specific areas, i would rather try to complete them here

**Pull request merged**